x-pack/plugins/global_search_bar/public/telemetry/event_reporter.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { METRIC_TYPE } from '@kbn/analytics';
+import { AnalyticsServiceStart } from '@kbn/core/public';
+import { UsageCollectionSetup } from '@kbn/usage-collection-plugin/public';
+import {
+  ClickMetric,
+  CountMetric,
+  EventMetric,
+  FieldType,
+  TrackedApplicationClick,
+  TrackedError,
+  TrackedSavedObjectClick,
+  TrackUiMetricFn,
+} from '../types';
+
+export class EventReporter {
+  private deps: {
+    analytics: AnalyticsServiceStart;
+  };
+  private trackUiMetric: TrackUiMetricFn;
+  private focusStart = Infinity;
+
+  constructor({
+    analytics,
+    usageCollection,
+  }: {
+    analytics: AnalyticsServiceStart;
+    usageCollection: UsageCollectionSetup | undefined;
+  }) {
+    this.deps = { analytics };
+
+    let trackUiMetric: TrackUiMetricFn = () => {};
+    if (usageCollection) {
+      trackUiMetric = (metricType, eventName, context) => {
+        let counter: [string, string | [string, string]];
+        switch (eventName) {
+          case ClickMetric.USER_NAVIGATED_TO_APPLICATION:
+          case ClickMetric.USER_NAVIGATED_TO_SAVED_OBJECT:
+            counter = [metricType, [eventName, `${eventName}_${context}`]];
+            break;
+          default:
+            // track simple UI Counter metrics
+            counter = [metricType, eventName];
+        }
+        usageCollection.reportUiCounter('global_search_bar', ...counter);
+      };
+    }
+
+    this.trackUiMetric = trackUiMetric;
+  }
+
+  /**
+   * Called when the text input component has the cursor focused
+   */
+  public searchFocus() {
+    this.trackUiMetric(METRIC_TYPE.COUNT, CountMetric.SEARCH_FOCUS);
+
+    this.focusStart = new Date(Date.now()).valueOf();
+  }
+
+  /**
+   * Called when the text input component has lost focus
+   */
+  public searchBlur() {
+    const focusTime = new Date(Date.now()).valueOf() - this.focusStart;
+    if (focusTime > 0) {
+      this.deps.analytics.reportEvent(EventMetric.SEARCH_BLUR, {
```

Comment:
nit: the same way you have an alias for `this.tracUiMetric` we could have an alias of `this.deps.analytics.reportEvent` in `this.reportEvent`?